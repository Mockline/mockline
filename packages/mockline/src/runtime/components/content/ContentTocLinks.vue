<script setup lang="ts">
import type { TocLink } from '@nuxt/content/dist/runtime/types'
import { useScrollspy, useNuxtApp, useRouter } from '#imports'

const { activeHeadings, updateHeadings } = useScrollspy()

export type ContentTocProps = {
  title?: string
  links: TocLink[]
  active?: string
}

withDefaults(defineProps<ContentTocProps>(), {
  title: 'Table of Contents',
  links: () => []
})

const nuxtApp = useNuxtApp()
const router = useRouter()

nuxtApp.hooks.hookOnce('page:finish', () => {
  updateHeadings([
    ...document.querySelectorAll('h2'),
    ...document.querySelectorAll('h3')
  ])
})

const emit = defineEmits(['move'])

const scrollToHeading = (id: string): void => {
  const encodedId = encodeURIComponent(id)
  router.push(`#${encodedId}`)
  emit('move', id)
}
</script>

<template>
  <ul v-if="links?.length">
    <li v-for="link in links" :key="link.text" :class="link.depth === 3 ? 'ml-3' : ''">
      <a
        class="block truncate text-sm/6"
        :class="activeHeadings.includes(link.id) ? 'text-[var(--ui-primary)]' : 'text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200'"
        :href="`#${link.id}`"
        @click.prevent="scrollToHeading(link.id)"
      >
        {{ link.text }}
      </a>

      <MContentTocLinks v-if="link.children" :links="link.children" />
    </li>
  </ul>
</template>
