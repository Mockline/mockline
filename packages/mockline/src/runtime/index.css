@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

