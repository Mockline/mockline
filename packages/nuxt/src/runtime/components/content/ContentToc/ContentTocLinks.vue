<script setup lang="ts">
import type { TocLink } from '@nuxt/content'
import { useRouter } from 'vue-router'

const { links = [] } = defineProps<{
  links: TocLink[]
  active?: string
  activeHeadings: string[]
}>()

const router = useRouter()

const scrollToHeading = (id: string): void => {
  const encodedId = encodeURIComponent(id)

  const targetElement = document.getElementById(id)
  if (targetElement) {
    const contentPanel = document.getElementById('app-main-content')
    if (contentPanel) {
      const targetPosition = targetElement.offsetTop

      contentPanel.scrollTo({
        top: targetPosition - 20,
        behavior: 'smooth'
      })
    }
  }
  router.push(`#${encodedId}`)
}
</script>

<template>
  <ul class="space-y-2">
    <li v-for="link in links" :key="link.text" :class="[link.depth === 3 ? 'ml-4' : '']">
      <a
        class="block truncate text-xs/4 transition-all duration-500 ease-in-out"
        :class="activeHeadings.includes(link.id) ? 'text-[var(--ui-primary)]' : 'text-[var(--ui-text-secondary)] hover:text-[var(--ui-text-primary)] dark:hover:text-[var(--ui-text-primary-dark)]'"
        :href="`#${link.id}`"
        @click.prevent="scrollToHeading(link.id)"
      >
        {{ link.text }}
      </a>

      <MContentTocLinks
        v-if="link.children"
        :links="link.children"
        :active-headings
      />
    </li>
  </ul>
</template>
