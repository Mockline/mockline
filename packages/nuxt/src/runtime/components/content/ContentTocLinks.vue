<script setup lang="ts">
import type { TocLink } from '@nuxt/content'
import { useRouter } from '#imports'

type ContentTocProps = {
  title?: string
  links: TocLink[]
  active?: string,
  activeHeadings: string[]
}
const { title = 'Table of Contents', links = [] } = defineProps<ContentTocProps>()

const router = useRouter()

const scrollToHeading = (id: string): void => {
  const encodedId = encodeURIComponent(id)
  router.push(`#${encodedId}`)
}
</script>

<template>
  <ul v-if="links?.length">
    <li v-for="link in links" :key="link.text" :class="link.depth === 3 ? 'ml-3' : ''">
      <a
        class="block truncate text-sm/6"
        :class="activeHeadings.includes(link.id) ? 'text-[var(--ui-primary)]' : 'text-gray-500 dark:text-gray-400 hover:text-gray-700 dark:hover:text-gray-200'"
        :href="`#${link.id}`"
        @click.prevent="scrollToHeading(link.id)"
      >
        {{ link.text }}
      </a>

      <MContentTocLinks v-if="link.children" :links="link.children" :active-headings />
    </li>
  </ul>
</template>
