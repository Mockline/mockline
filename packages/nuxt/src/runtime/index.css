@import "@mockline/themes";
@import "./mockline.css";
