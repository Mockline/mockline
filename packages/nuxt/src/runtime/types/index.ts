// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
