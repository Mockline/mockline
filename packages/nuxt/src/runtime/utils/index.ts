export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
