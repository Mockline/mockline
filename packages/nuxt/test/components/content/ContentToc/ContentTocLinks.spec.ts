import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import ContentTocLinks from '../../src/runtime/components/content/ContentToc/ContentTocLinks.vue'

describe('ContentTocLinks', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(ContentTocLinks)
    expect(html.html()).toMatchSnapshot()
  })
})
