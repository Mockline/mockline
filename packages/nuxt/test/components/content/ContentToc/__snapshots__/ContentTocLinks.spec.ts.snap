// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentTocLinks > can mount component 1`] = `
"<div id="test-wrapper">
  <ul class="space-y-2"></ul>
</div>"
`;
